x-pack/plugins/ml/public/jobs/components/custom_url_editor/utils.js

```diff
@@ -175,18 +175,18 @@ function buildDashboardUrlFromSettings(settings) {
         // which includes the ID of the index holding the field used in the filter.
 
         // So for simplicity, put entities in the query, replacing any query which is there already.
-        // e.g. query:(language:lucene,query:'region:us-east-1%20AND%20instance:i-20d061fa')
+        // e.g. query:(language:kuery,query:'region:us-east-1%20and%20instance:i-20d061fa')
```

Comment:
as these blocks of code are identical, is it worth moving them to a `createQueryString` function?

Reply:
Created a `buildAppStateQueryParam` function for use by the functions which create the Discover and Dashboard URLs